project/src/main/webapp/script.js

```diff
@@ -400,3 +407,29 @@ function registerUserByToken() {
   }
   fetch('/register?idToken=' + googleUser.getAuthResponse().id_token, {method: 'POST'});
 }
+
+/** Sends feedback that includes the place the user chose. */
+function sendUserChoiceAsFeedback() {
+  let indexOfPlaceUserChose = getCheckedValueByElementId('chosen-place-form',
+      'Please select the place that you chose.');
+  if (indexOfPlaceUserChose <= recommendedPlaces.length) {
+    storeUserFeedback(recommendedPlaces, recommendedPlaces[indexOfPlaceUserChose].placeId,
+        false /** user tried again */)
+  }
+}
+
+/** Stores user feedback in the database, by using the feedback servlet. */
+function storeUserFeedback(recommendedPlaces, chosenPlaceId, tryAgain) {
```

Comment:
Maybe clear "recommendedPlaces" after you send the feedback?

Reply:
done